kb/2023-02-01/scan_and_repair_filesystem.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Scan and Repair Root Filesystem of VirtualMachine
+description: Scan and repair root filesystem of VM
+slug: scan-and-repair-vm-root-filesystem
+authors:
+  - name: Vicente Cheng
+    title: Senior Software Engineer
+    url: https://github.com/Vicente-Cheng
+    image_url: https://github.com/Vicente-Cheng.png
+tags: [storage, longhorn, root, filesystem]
+hide_table_of_contents: false
+---
+
+In the early version of harvester (v1.0.3 and before versions), the longhorn volume sometimes corrupts when rebuilt (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In the harvester v1.1.0 and later versions, the longhorn team already fix this problem. Therefore, you could scan your filesystem of the VM by following the steps and repairing it (if needed).
+
+## Stop The VM And Backup Volume
+
+Before we scan the filesystem, we recommend you back up your volume. For example, refer to the following steps to stop the VM and backup the volumes.
+
+- Find the target VM
+
+![finding the target VM](./imgs/finding_the_target_vm.png)
+
+- Stop the target VM
+
+![Stop the target VM](./imgs/stop_the_target_vm.png)
+
+Now, the VM stopped, and related volumes are detached. Then we go to the longhorn UI to backup this volume.
+
+- Find the related volume of VM. Then, attach it again for scanning.
```

Comment:
Need a precise way to find the volume. One way is (for v1.1.1):
- Enable "Developer Tools & Features" (like https://docs.harvesterhci.io/v1.1/troubleshooting/harvester#access-embedded-rancher-and-longhorn-dashboards)
- Click VM for detail.
- Click the Volumes tab. There is an arrow icon called "check volume details", that should lead to LH volume page.